lib/bank_account/savings_account.rb

```diff
@@ -0,0 +1,28 @@
+module Bank
+  class SavingsAccount < Account
+
+    def initialize(id, initial_balance)
+      super
+      raise ArgumentError.new "Insufficient Funds" unless initial_balance >= 10
+    end
+
+    # charges $2 for every withdrawal
+    def withdraw(amount)
+      @balance -= (amount + 2)
+      if @balance < 10
+        puts "Savings can not fall below $10.Your remaining savings amount is: "
+        return @balance += (amount + 2)
+      else
+        return @balance
+      end
+    end
+
+    # calculates interest on balance where rate is in %
+    def add_interest(rate)
+      @accrued_interest = @balance * (rate/100.to_f)
+      @balance += @accrued_interest
+      return @accrued_interest
+    end
```

Comment:
I remember having trouble getting the calculation to work because it kept returning 0 so I added the float conversion and that seemed to help. After that, I didn't touch it again because it was working.